pkg/lib/infra/sms/sender.go

```diff
@@ -9,16 +9,14 @@ import (
 
 type Logger struct{ *log.Logger }
 
-func NewLogger(lf *log.Factory) Logger { return Logger{lf.New("sms-client")} }
+func NewLogger(lf *log.Factory) Logger { return Logger{lf.New("sms-sender")} }
 
-type Client struct {
+type Sender struct {
 	Logger         Logger
 	ClientResolver *ClientResolver
 }
 
-var _ smsapi.Client = (*Client)(nil)
-
-func (c *Client) Send(ctx context.Context, opts smsapi.SendOptions) error {
+func (c *Sender) Send(ctx context.Context, client smsapi.Client, opts smsapi.SendOptions) error {
 	client, _, err := c.ClientResolver.ResolveClient()
```

Comment:
I think we should use the client in the argument list.

Reply:
Right, fixed thanks!